csharp.test/TestMaps.cs

```diff
@@ -190,7 +199,7 @@ private static void DoRoundtripTest(bool optional, string[][] keys, string[][] v
                 {
                     using var propertiesBuilder = new WriterPropertiesBuilder();
                     using var writerProperties = propertiesBuilder.Build();
-                    using var schemaNode = CreateMapSchema(optional);
+                    schemaNode ??= CreateMapSchema(optional);
```

Comment:
This is getting quite nitpicky, but it might be cleaner to always require the schema node to be passed in, then the `optional` parameter can be removed from `DoRoundtripTest`, and rather than creating the map schema inline in the call to `DoRoundtripTest` we can create it with a `using` statement to make sure the native handle is always freed immediately rather than requiring the finalizer to be called.